### DISCUSSION TITLE
Endpickevent

### DISCUSSION THREAD

**Discussion by HametAl** [pyvista/pyvista#4209] [Q&A] [answered]
How can I assign an observer for end  pick event. I need to run a function once I unpick or deselect an actor. Is this possible?

**Comment by banesullivan** [comment, marked as answer]
You can grab the picker and add an observer for that event:

```py
import pyvista as pv
import vtk

pl = pv.Plotter(notebook=False)
pl.add_mesh(pv.Sphere())
pl.enable_mesh_picking()

picker = pl.iren.get_picker()
picker.AddObserver(vtk.vtkCommand.EndPickEvent, callback)

pl.show()
```

I am actively refactoring all of the picking utilities and the picking API will go under a major change. I'll keep this use case in mind try to implement something better.

**Reply by HametAl** [reply]
Thanks! It worked perfectly. Was exactly what I was looking for.
